Lib/test/test_builtin.py

```diff
@@ -340,11 +340,10 @@ def test_compile(self):
         self.assertRaises(TypeError, compile)
         self.assertRaises(ValueError, compile, 'print(42)\n', '<string>', 'badmode')
         self.assertRaises(ValueError, compile, 'print(42)\n', '<string>', 'single', 0xff)
-        self.assertRaises(ValueError, compile, chr(0), 'f', 'exec')
         self.assertRaises(TypeError, compile, 'pass', '?', 'exec',
                           mode='eval', source='0', filename='tmp')
         compile('print("\xe5")\n', '', 'exec')
-        self.assertRaises(ValueError, compile, chr(0), 'f', 'exec')
+        self.assertRaises(SyntaxError, compile, chr(0), 'f', 'exec')
```

Comment:
Is this change originated from CPython? Otherwise it needs to be reverted too

Reply:
Yes - from the last commit message, it seems that the tests were copied from CPython 3.8. But in CPython3.12, the test is slightly different, with a SyntaxError instead:

[https://github.com/python/cpython/blob/30725f0ad207511ca146bc8130a10513fbed7a7b/Lib/test/test_builtin.py#L349](https://github.com/python/cpython/blob/30725f0ad207511ca146bc8130a10513fbed7a7b/Lib/test/test_builtin.py#L349)

Reply:
Thank you for confirming!